src/main/java/de/tum/in/www1/artemis/security/jwt/JWTFilter.java

```diff
@@ -28,23 +30,32 @@ public JWTFilter(TokenProvider tokenProvider) {
     @Override
     public void doFilter(ServletRequest servletRequest, ServletResponse servletResponse, FilterChain filterChain) throws IOException, ServletException {
         HttpServletRequest httpServletRequest = (HttpServletRequest) servletRequest;
-        String jwt = resolveToken(httpServletRequest);
-        if (StringUtils.hasText(jwt) && this.tokenProvider.validateTokenForAuthority(jwt)) {
-            Authentication authentication = this.tokenProvider.getAuthentication(jwt);
-            SecurityContextHolder.getContext().setAuthentication(authentication);
+        Cookie jwtCookie = WebUtils.getCookie(httpServletRequest, JWT_COOKIE_NAME);
+        if (jwtCookie != null) {
+            String jwt = jwtCookie.getValue();
+            if (StringUtils.hasText(jwt) && this.tokenProvider.validateTokenForAuthority(jwt)) {
+                Authentication authentication = this.tokenProvider.getAuthentication(jwt);
+                SecurityContextHolder.getContext().setAuthentication(authentication);
+            }
         }
         filterChain.doFilter(servletRequest, servletResponse);
     }
 
-    private String resolveToken(HttpServletRequest request) {
-        String bearerToken = request.getHeader(AUTHORIZATION_HEADER);
-        if (StringUtils.hasText(bearerToken) && bearerToken.startsWith("Bearer ")) {
-            return bearerToken.substring(7);
-        }
-        String jwt = request.getParameter(AUTHORIZATION_TOKEN);
-        if (StringUtils.hasText(jwt)) {
-            return jwt;
-        }
-        return null;
+    /**
+     * Builds the cookie containing the jwt
+     * @param jwt      the token that will be used as the cookie's value
+     * @param duration the validity of the cookie
+     * @return         the response cookie that should be set containing the jwt
+     */
+    public static ResponseCookie buildJWTCookie(String jwt, Duration duration) {
+        // @formatter:off
```

Comment:
In some places (e.g. ProgrammingExerciseGradingServiceTest.java) we use empty comments after each line to keep the line changes even with the formatter on. You might want to consider/unify this